### PR TITLE
Blog App: Add Form

### PR DESCRIPTION
**Pull request by sja-thedude**
- Added all necessary routing for creating posts, comments, and likes.
- Added button in the home page "Add new post" (for the current user)
- Added Post creation (form, style, and logic)
- Added Comment creation (form, style, and logic)
- Added Like creation (form, style, and logic)

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      user_id: current_user.id,
+      post_id: @post.id
+    )
+    new_like.update_likes_counter
+    if new_like.save
+      redirect_to "/users/#{@post.user_id}/posts/#{@post.id}", notice: 'Success!'
+    else
+      render :new, alert: 'Error occured!'
+    end
+  end
```
Comment (Emmyn5600) [inline app/controllers/likes_controller.rb]: - [x] Kindly, fix the create method for the likes, when a user clicks on the like button the likes are being counted however it is causing this issue, please fix it.

image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      render :new, alert: 'Error occured!'
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)[:text]
+  end
+end
```
Comment (Emmyn5600) [inline app/controllers/comments_controller.rb:22]: - [x] The comment text area is only appearing to the two users `Ben and Tailah` while for others it is not appearing. Kindly make sure that all users can see the comment text area.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @sja-thedude 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet, please work on it and ask for another review.

- You have a descriptive PR
- You have the descriptive Readme file.

 ## Required Changes ♻️
 
 - Check the comment under the review too.
 - Make more descriptive commits messages. 
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (sja-thedude) [reply app/controllers/likes_controller.rb]: Thanks sir, its fixed

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (sja-thedude) [reply app/controllers/comments_controller.rb:22]: Okay sir

**Review by t-yanick — APPROVED** [review]
Hi @sja-thedude, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**